vpr/src/route/rr_graph_storage.h

```diff
@@ -585,7 +616,50 @@ class t_rr_graph_storage {
                             "Attempted to access RR node 'side' for non-IPIN/OPIN type '%s'",
                             rr_node_typename[node_data.type_]);
         }
-        return node_storage[id].dir_side_.side;
+        std::bitset<NUM_SIDES> side_tt = node_storage[id].dir_side_.sides;
+        for (const e_side& side : SIDES) {
+            if (side_tt[size_t(side)]) {
+                return side;
+            }
+        }
+        /* No valid sides, return an invalid value */
+        return NUM_SIDES;
```

Comment:
This should throw an exception or call `VTR_FATAL_ERROR`?

Reply:
No problem. Should throw a fatal error as other methods